### ISSUE TITLE
DataTable: Cell Editing can't clear value

### ISSUE THREAD

**Issue by cutie77** [primefaces/primereact#4552]
### Describe the bug

Hello, 

I am using primereact datatable cell editing but I am not able to delete entire value from the cell. You can adjust it but you can't delete it. When you try to delete entire value it is automatically returned back. Could you please fix that problem?

Thank you.

### Reproducer

 https://stackblitz.com/edit/react-9j198l 

### PrimeReact version

9.5.0

### React version

18.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

**Comment by melloware** [comment]
Can you update this reproducer to show the issue please: https://stackblitz.com/edit/react-9j198l ?

**Comment by melloware** [comment]
Actually it looks like it happens in that reproducer.

**Comment by cutie77** [comment]
Right, if you try to delete some value in the datatable in that reproducer you just can't delete it because it will return you back the same value.

**Comment by ubau** [comment]
It does not work for "Price" and "Quantity" because `onCellEditComplete` callback func  checks if `newVal` is positive number and for other fields it checks if newVal data length > 0. If you remove these restrictions then it works as you expected.

**Comment by ubau** [comment]
And you need a bit extend "Price" template function by checking for `null` and returning empty string in such case.

**Comment by melloware** [comment]
@cutie77 see @ubau answer he is correct. It works if you remove the data checking in that example.

**Comment by cutie77** [comment]
yes if you adjust restrictions it works. Thank you, I didn't notice
